network/ServerNetworking.cpp

```diff
@@ -30,17 +30,24 @@ namespace {
 class DiscoveryServer {
 public:
     DiscoveryServer(boost::asio::io_context& io_context) :
-        m_socket(io_context, udp::endpoint(udp::v6(), Networking::DiscoveryPort()))
+        m_service(io_context),
+        m_socket(io_context)
     { Listen(); }
 
 private:
     void Listen() {
-        m_recv_buffer.fill('\0');
-        m_socket.async_receive_from(
-            boost::asio::buffer(m_recv_buffer),
-            m_remote_endpoint,
-            boost::bind(&DiscoveryServer::HandleReceive, this,
-                        boost::asio::placeholders::error));
+        // use a dual stack (ipv6 + ipv4) socket
+        udp::endpoint discovery_endpoint(udp::v6(), Networking::DiscoveryPort());
+
+        if (!GetOptionsDB().Get<bool>("singleplayer")) {
```

Comment:
Why? This would prevent querying a single player server for its status from the same machine.

Reply:
Should single player server be answering to that queries?

Reply:
Why not?

Reply:
Made it listen loopback in singleplayer mode.